src/Carbon/CarbonInterval.php

```diff
@@ -181,38 +181,9 @@ public static function create($years = 1, $months = null, $weeks = null, $days =
      */
     public static function __callStatic($name, $args)
     {
-        $arg = count($args) === 0 ? 1 : $args[0];
-
-        switch ($name) {
-            case 'years':
-            case 'year':
-                return new static($arg);
-
-            case 'months':
-            case 'month':
-                return new static(null, $arg);
-
-            case 'weeks':
-            case 'week':
-                return new static(null, null, $arg);
-
-            case 'days':
-            case 'dayz':
-            case 'day':
-                return new static(null, null, null, $arg);
-
-            case 'hours':
-            case 'hour':
-                return new static(null, null, null, null, $arg);
-
-            case 'minutes':
-            case 'minute':
-                return new static(null, null, null, null, null, $arg);
-
-            case 'seconds':
-            case 'second':
-                return new static(null, null, null, null, null, null, $arg);
-        }
+        return call_user_func_array(
+            array(new static(0), $name), $args
+        );
```

Comment:
@imbrish This is shorter code but it means construct + call instead of just construct. It rely on the fluent setter when previously not needed. This means this could easily break some CarbonInterval sub-class.